src/datasources/__tests__/grid.test.js

```diff
@@ -0,0 +1,290 @@
+              LongName: 'grid_name'
+            }
+          }]
+        })
+      const response = await gridDatasource({}, { headers: { 'CMR-Request-Id': 'abcd-1234-efgh-5678' } }, requestInfo)
+
+      expect(response).toEqual({
+        count: 84,
+        cursor: 'eyJ1bW0iOiJbXCJ4eXpcIiwgNzg5LCA5OTldIn0=',
+        items: [{
+          conceptId: 'GRD100000-EDSC',
+          longName: 'grid_name'
+        }]
+      })
+    })
+
+    describe('when a cursor is requested', () => {
+      test('requests a cursor', async () => {
+        nock(/example/)
+          .defaultReplyHeaders({
+            'CMR-Hits': 84,
+            'CMR-Took': 7,
+            'CMR-Request-Id': 'abcd-1234-efgh-5678',
+            'CMR-Search-After': '["xyz", 789, 999]'
+          })
+          .post(/grids\.umm_json/)
+          .reply(200, {
+            items: [{
+              meta: {
+                'concept-id': 'GRD100000-EDSC'
+              },
+              umm: {
+                LongName: 'grid_name'
+              }
+            }]
+          })
+
+        const response = await gridDatasource({}, { headers: { 'CMR-Request-Id': 'abcd-1234-efgh-5678' } }, requestInfo)
+
+        expect(response).toEqual({
+          count: 84,
+          cursor: 'eyJ1bW0iOiJbXCJ4eXpcIiwgNzg5LCA5OTldIn0=',
+          items: [{
+            conceptId: 'GRD100000-EDSC',
+            longName: 'grid_name'
+          }]
+        })
+      })
+    })
+  })
+
+  describe('without params', () => {
+    test('returns the parsed grid results', async () => {
+      nock(/example/)
+        .defaultReplyHeaders({
+          'CMR-Hits': 84,
+          'CMR-Took': 7,
+          'CMR-Request-Id': 'abcd-1234-efgh-5678'
+        })
+        .post(/grids\.json/)
+        .reply(200, {
+          items: [{
+            concept_id: 'GRD100000-EDSC'
+          }]
+        })
+
+      const response = await gridDatasource({}, { headers: { 'CMR-Request-Id': 'abcd-1234-efgh-5678' } }, requestInfo)
+
+      expect(response).toEqual({
+        count: 84,
+        cursor: null,
+        items: [{
+          conceptId: 'GRD100000-EDSC'
+        }]
+      })
+    })
+  })
+
+  describe('with params', () => {
+    test('returns the parsed grid results', async () => {
+      nock(/example/)
+        .defaultReplyHeaders({
+          'CMR-Hits': 84,
+          'CMR-Took': 7,
+          'CMR-Request-Id': 'abcd-1234-efgh-5678'
+        })
+        .post(/grids\.json/, 'concept_id=GRD100000-EDSC')
+        .reply(200, {
+          items: [{
+            concept_id: 'GRD100000-EDSC'
+          }]
+        })
+
+      const response = await gridDatasource({ params: { concept_id: 'GRD100000-EDSC' } }, { headers: { 'CMR-Request-Id': 'abcd-1234-efgh-5678' } }, requestInfo, 'grid')
+
+      expect(response).toEqual({
+        count: 84,
+        cursor: null,
+        items: [{
+          conceptId: 'GRD100000-EDSC'
+        }]
+      })
+    })
+  })
+
+  describe('with only umm keys', () => {
```

Comment:
In the collection and granule datasource tests I see a test `with json and umm keys`. I'm not sure why that test is missing in the other datasource tests, but it feels like a valuable test that should be added.

Reply:
I think the test isn't valid for some of the concepts. In the Granule and Collection test both are using one field which is shared, one which is returned from the json endpoint, and one from the umm_json endpoint. For some of the concepts like tools, variables, and services there are no types in the type definition file which are returned only through the .json endpoint, except the ones which are shared between .json and umm_json